### PR TITLE
Look at logging verbosity in addition to test verbosity.

### PR DESCRIPTION
**Pull request by mberhault**
This allows --vmodule=logic_test=1

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/3368)

### CODE REVIEW

File: sql/logic_test.go [hunk 1]
```diff
@@ -401,7 +402,7 @@ SET DATABASE = test;
 }
 
 func (t *logicTest) execStatement(stmt logicStatement) {
-	if testing.Verbose() {
+	if testing.Verbose() || log.V(1) {
```
Comment (bdarnell) [inline sql/logic_test.go]: Do we need to look at both? I think I would just use `log.V()`. `testing.Verbose()` is kind of weird; it means several things including "print output even on success", so we turn it on in our CI builds whether we want to increase the verbosity of the logs or not.

**Comment by tamird** [comment]
This doesn't quite work as expected; without `-v` the built-in buffering prevents anything from being printed unless the test fails. This might be OK for this use case, but still a bit surprising. Can you add a comment that explains this?

LGTM

File: sql/logic_test.go (hunk 1 above)
Comment (mberhault) [inline sql/logic_test.go]: Yeah, but I think we intentionally use that in some places.

**Comment by mberhault** [comment]
I tested the `sql.test` executable and with `--vmodule=logic_test=1`, I got the queries logged to stdout all the time

**Comment by tamird** [comment]
Weird. Does `make test PKG=./sql TESTFLAGS='--vmodule=logic_test=1' TESTS=Logic` log the queries? it doesn't on my machine.

**Comment by mberhault** [comment]
if you pass `-v` to the test, that will work. `go test` hides stderr/stdout for successful runs, but the built test binary does not. odd

**Comment by mberhault** [comment]
so Ben's suggestion of dropping `testing.Verbose` entirely might be best. `sql.test` will still output things properly with `--vmodule`, and `go test` or `make test` will output debug logs with `-v` as usual, and can take `--vmodule` to also log statements.
The only reason to keep the `||` as I have it here would be to avoid having to say `--vmodule` when `-v` is also specified. I'm cool either way.

**Comment by mberhault** [comment]
@tamird, @bdarnell: what's the verdict? do we want `go test -v` to show those or always require `--vmodule`?

**Comment by petermattis** [comment]
I use `go test -v -run TestLogic` frequently.

**Comment by mberhault** [comment]
ok, then I'll keep the change as posted here

**Comment by bdarnell** [comment]
LGTM

**Pull request merged**